learntools/core/tracking.py

```diff
@@ -0,0 +1,56 @@
+import enum
+from IPython.display import display, Javascript
+import json
+
+import learntools
+
+# If set to True, then echo logged events as output.
+DEBUG = False
+
+class InteractionType(enum.Enum):
+    CHECK = 1
+    HINT = 2
+    SOLUTION = 3
+
+class OutcomeType(enum.Enum):
+    PASS = 1
+    FAIL = 2
+    EXCEPTION = 3
+    UNATTEMPTED = 4
+
+_EVENT_DEFAULTS = dict(
+        learnToolsVersion = str(learntools.__version__),
+        valueTowardsCompletion = 0.0,
+        failureMessage = '',
+        exceptionClass = '',
+        trace = '',
+)
+
+def track(event):
+    # TODO: could be nice to put some validation logic here.
+    for k, v in _EVENT_DEFAULTS.items():
+        event.setdefault(k, v)
+
+    # Convert enum values to plain ints
+    intxn_type = event['interactionType']
```

Comment:
I tend to be pretty meticulous about using variables for any expression I'm going to be referring to more than once. Happy to rename though.

Reply:
I would think the value of naming depends on the complexity of the expression and the frequency with which it is used, and this might be an exception on those criteria.  But it's not a big deal to me to keep it.

I would find `interaction_type` easier to read than `intxn_type`.